var/spack/repos/builtin/packages/h5utils/package.py

```diff
@@ -12,10 +12,18 @@ class H5utils(AutotoolsPackage):
 
     homepage = "http://ab-initio.mit.edu/wiki/index.php/H5utils"
     url      = "http://ab-initio.mit.edu/h5utils/h5utils-1.12.1.tar.gz"
-    list_url = "http://ab-initio.mit.edu/h5utils/old/"
+    list_url = 'https://github.com/NanoComp/h5utils'
 
+    version('1.13.1', sha256='c5a76f064d6daa3e65583dce2b61202510e67cf6590f076af9a8aa72511d7d65')
     version('1.12.1', sha256='7290290ca5d5d4451d757a70c86baaa70d23a28edb09c951b6b77c22b924a38d')
 
+    def url_for_version(self, version):
+        """The release of 1.12.1 isn't on github but on ab-initio, using self.url"""
+        url = 'https://github.com/NanoComp/h5utils/releases/download/{0}/h5utils-{0}.tar.gz'
+        if (version == Version('1.12.1')):
+            return self.url
+        return url.format(version)
```

Comment:
And use a secure location for v1.12.1 plus simplify the package with the above `url` fix PLUS:

```suggestion

    version('1.13.1', sha256='c5a76f064d6daa3e65583dce2b61202510e67cf6590f076af9a8aa72511d7d65')
    version('1.12.1', sha256='7290290ca5d5d4451d757a70c86baaa70d23a28edb09c951b6b77c22b924a38d',
            url = "https://github.com/NanoComp/h5utils/archive/refs/tags/1.12.1.tar.gz")
```

Reply:
Otherwise, at least move the `url_for_version` method to the bottom of the file.